capstone/backend/src/main/java/com/google/servlets/StoreRankingsServlet.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2020 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.servlets;
+
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestFactory;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.cloud.spanner.SpannerException;
+import com.google.gson.Gson;
+import com.google.spanner.LibraryFunctions;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/api/v1/get-store-rankings")
+public class StoreRankingsServlet extends HttpServlet {
```

Comment:
Do we need a new servlet? Is `GetStoresWithItemTypesServlet` insufficient?

Reply:
I modified the `GetStoresWithItemTypesServlet` and changed the name to `StoreRankingsServlet` because I thought it fit better. I kept the functionality from the original `GetStoresWithItemTypesServlet` and integrated the distance api to get to get the distance from the user to each of the stores. Using that, I can filter out stores that are too far away and better calculate a storeScore to rank the stores.

Reply:
Sorry, i didn't realize you deleted the other file.